cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/Schedulers.java

```diff
@@ -136,4 +131,25 @@ public static String getQuartzCronExpression(String cronEntry) {
       return cronEntry;
     }
   }
+
+  /**
+   * Adds datasets and types to the given {@link DatasetFramework}. Used by the upgrade tool to upgrade Datasets
+   *
+   * @param datasetFramework framework to add types and datasets to
+   */
+  public static void setupJobQueueDataset(DatasetFramework datasetFramework) throws IOException,
```

Comment:
why put it here instead of in the JobQueueDataset class like we did for all other datasets?

Reply:
Its seems like `Schedulers` wraps both jobqueue and schedule store datasets and `JobQueueDataset` only has serde operations. If you think this placement does not make sense, I can move it to `JobQueueDataset`

Reply:
hm I see. I don't think those methods should be in Schedulers either though there is no need to move them out right now. I do think the setup methods should be in the dataset class, as that is the pattern that we have established elsewhere.